pkg/common/resource_test.go

```diff
@@ -243,3 +244,76 @@ func TestNodeResource(t *testing.T) {
 
 	assert.Equal(t, result.Resources[CPU].GetValue(), int64(14500))
 }
+
+func TestIsZero(t *testing.T) {
+	r := NewResourceBuilder().
+		AddResource(Memory, 1).
+		AddResource(CPU, 1).
+		Build()
+	assert.Equal(t, IsZero(r), false)
+
+	r = NewResourceBuilder().
+		AddResource(CPU, 0).
+		Build()
+	assert.Equal(t, IsZero(r), true)
+
+	r = NewResourceBuilder().
+		AddResource(Memory, 0).
+		AddResource(CPU, 0).
+		Build()
+	assert.Equal(t, IsZero(r), true)
+
+	r = NewResourceBuilder().
+		AddResource(Memory, 0).
+		AddResource(CPU, 1).
+		Build()
+	assert.Equal(t, IsZero(r), false)
+
+	assert.Equal(t, IsZero(nil), true)
+
+	r = &si.Resource{}
+	assert.Equal(t, IsZero(r), true)
+}
+
+func TestSub(t *testing.T) {
+	// simple case (nil checks)
+	result := Sub(nil, nil)
+	if result == nil || len(result.Resources) != 0 {
+		t.Errorf("sub nil resources did not return zero resource: %v", result)
+	}
+
+	// empty resources
+	left := NewResourceBuilder().Build()
+	result = Sub(left, nil)
+	if result == nil || len(result.Resources) != 0 || result == left {
+		t.Errorf("sub Zero resource (right) did not return cloned resource: %v", result)
+	}
+
+	// simple empty resources
+	res1 := NewResourceBuilder().
+		AddResource("a", 5).
+		Build()
+	result = Sub(left, res1)
+	assert.Equal(t, IsZero(result), true)
+
+	// complex case: just checking the resource merge, values check is secondary
+	res1 = NewResourceBuilder().
+		AddResource("a", 0).
+		AddResource("b", 1).
+		Build()
+	res2 := NewResourceBuilder().
+		AddResource("a", 1).
+		AddResource("c", 0).
+		AddResource("d", -1).
+		Build()
+	res3 := Sub(res1, res2)
+
+	expected := NewResourceBuilder().
+		AddResource("a", -1).
+		AddResource("b", 1).
+		Build()
```

Comment:
That is not what I would expect based on the way we do *add*, see comment in implementation code